### PR TITLE
Snap back to initial center after delay

### PR DESCRIPTION
**Pull request by mattThousand**
- After a time interval specified by a new constructor parameter (`duration`, defaults to `1.0`), a `UISnapBehavior is applied to each subview, snapping it back into place

### CODE REVIEW

File: FlubberView Example/ViewController.swift [hunk 1]
```diff
@@ -169,7 +169,7 @@ class ViewController: UIViewController {
         }
 
         frequencySlider.callBack = { frequency in
-            self.flubberView.frequency = frequency
+            self.flubberView.frequency = frequency * 5.0
```
Comment (ateliercw) [inline FlubberView Example/ViewController.swift]: #pp change the constants rather than just multiplying them

File: FlubberView/FlubberView.swift [hunk 2]
```diff
@@ -110,22 +111,27 @@ public extension FlubberView {
...
             if let behavior = oldBehavior {
                 mainAnimator.removeBehavior(behavior)
             }
 
-            mainAnimator.addBehavior(snapBehavior)
-            v.center = CGPoint(x: v.center.x <~> elasticity, y: v.center.y <~> elasticity)
+            mainAnimator.addBehavior(bounceBehavior)
+            v.center = CGPoint(x: v.center.x <~> magnitude.elasticity, y: v.center.y <~> magnitude.elasticity)
             mainAnimator.updateItem(usingCurrentState: v)
+            DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + duration, execute: {
```
Comment (ateliercw) [inline FlubberView/FlubberView.swift:139]: #pp debounce this so that if the flubber view is tapped repeatedly, only the last snap back fires

**Review by ateliercw — APPROVED** [review]
👍 solid improvement to the animation behavior

**Comment by mattThousand** [comment]
@ateliercw thanks for the feedback! ready for another look

File: FlubberView/FlubberView.swift [hunk 3]
```diff
@@ -126,10 +129,18 @@ public extension FlubberView {
...
             mainAnimator.updateItem(usingCurrentState: v)
+
+            // Debounce snap behavior
+            if let snap = snapBehavior {
+                mainAnimator.removeBehavior(snap)
+            }
+
+            // Snap back to original position
             DispatchQueue.main.asyncAfter(deadline: DispatchTime.now() + duration, execute: {
                 let snap = UISnapBehavior(item: v, snapTo: initialPoint)
```
Comment (ateliercw) [inline FlubberView/FlubberView.swift:140]: #req I started teasing this apart, and the snap behavior is being set for every point, and the operation still fires every time, the snap behaviors need to be stored in a map table like the attachment behaviors, and then the update needs to be scheduled at the end of the loop for every snap operation inside an Operation that is canceled if the view is clicked again.

**Pull request closed without merging**